zio-http/shared/src/main/scala/zio/http/endpoint/Endpoint.scala

```diff
@@ -612,7 +612,7 @@ final case class Endpoint[PathInput, Input, Err, Output, Auth <: AuthType](
       output = ((HttpCodec.content[Output2](mediaType) ++ StatusCodec.status(status)) ?? doc) | self.output,
       error,
       codecError,
-      doc,
```

Comment:
This was not using the correct `doc`. It was using the one passed in parameter and not the one from `this`